fhir-server/src/main/java/com/ibm/fhir/server/util/IssueTypeToHttpStatusMapper.java

```diff
@@ -36,15 +43,20 @@
             for (Issue issue : issues) {
                 IssueType code = issue.getCode();
                 if (code != null && code.getValue() != null) {
-                    return issueTypeToResponseCode(IssueType.ValueSet.from(code.getValue()));
+                    ValueSet issueType = IssueType.ValueSet.from(code.getValue());
+                    // Special case for IssueType CONFLICT which can be either an HTTP 409 (Conflict) or HTTP 412 (Precondition failed)
+                    if (issueType == IssueType.ValueSet.CONFLICT &&
+                            FHIRUtil.getExtensionStringValue(code, EXTENSION_URL_HTTP_FAILED_PRECONDITION) != null) {
+                        return Status.PRECONDITION_FAILED;
+                    }
+                    return issueTypeToResponseCode(issueType);
                 }
             }
         }
-        
         return Status.INTERNAL_SERVER_ERROR;
     }
     
-    public static Status issueTypeToResponseCode(IssueType.ValueSet value) {
+    private static Status issueTypeToResponseCode(IssueType.ValueSet value) {
```

Comment:
any reason to 'lock' it down?

Reply:
I added special logic into the other method (which calls this one).  If you called this one directly (e.g. by accident) then you'd miss out on that new logic.